### PR TITLE
CommitSig Serialization refactor

### PR DESCRIPTION
**Pull request by greg-szabo**
Issue #247 - serialization refactor CommitSig "blob"
While chopping up PR #248 into multiple smaller PRs, this one that implements the enum-based CommitSig derailed from master so much that it's easier to cherry-pick the commits from this and open a new PR. That's what's going to happen. I'm keeping this open and in draft state until we were able to merge all changes from this into master using these smaller PRs. Then I'll close it.

In this step the CommitSig enum is introduced and the serializers.rs file is moved into its own folder and broken out into multiple files.

Unfortunately I couldn't separate the file move from the implementation of the CimmitSig. (Originally, the steps were 1. interim CommitSig implementation 2. move to folder 3. CommitSig enum implementation and cherry-picking the second step didn't work well without using the first step, but that's not needed any more.)

Converting this into a draft as long as step 2 (#269) is not merged.

* [X] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [X] Updated all code comments where relevant
* [X] Wrote tests
* [X] Updated CHANGES.md

### CODE REVIEW

File: tendermint/src/block/commit_sig.rs [hunk 1]
```diff
@@ -1,78 +1,100 @@
...
+        match value.block_id_flag {
+            BlockIDFlag::BlockIDFlagAbsent => {
+                if value.timestamp.is_some() {
+                    return Err("timestamp is present for BlockIDFlagAbsent CommitSig");
+                }
+                if value.signature.is_some() {
+                    return Err("signature is present for BlockIDFlagAbsent CommitSig");
+                }
+                Ok(CommitSig::BlockIDFlagAbsent {
+                    validator_address: value.validator_address,
```
Comment (melekes) [inline tendermint/src/block/commit_sig.rs:69]: where's validator_address validation?

File: tendermint/src/lite_impl/signed_header.rs [hunk 2]
```diff
@@ -59,8 +60,18 @@ impl lite::Commit for block::signed_header::SignedHeader {
             );
         }
 
+        // TODO: this last check is only necessary if we do full verification (2/3)
```
Comment (melekes) [inline tendermint/src/lite_impl/signed_header.rs:63]: why it's a TODO?

Comment (greg-szabo) [reply tendermint/src/lite_impl/signed_header.rs:63]: No idea, it was in the code from before.

**Comment by greg-szabo** [comment]
Closing in favor of #277 . Anton's notes were carried over too.

**Pull request closed without merging**